packages/xdl/src/Project.ts

```diff
@@ -490,10 +490,14 @@ export async function exportForAppHosting(
     const androidMapPath = path.join(outputDir, 'bundles', androidMapName);
     await writeArtifactSafelyAsync(projectRoot, null, androidMapPath, androidSourceMap);
 
-    // Remove original mapping to incorrect sourcemap paths
-    logger.global.info('Configuring sourcemaps');
-    await truncateLastNLines(iosJsPath, 1);
-    await truncateLastNLines(androidJsPath, 1);
+    if (!(target === 'bare' && semver.lte('40.0.0', exp.sdkVersion))) {
```

Comment:
```suggestion
    if (target === 'managed' && semver.lte('39.0.0', exp.sdkVersion)) {
```

imo this is a bit easier to read

Reply:
this changes the logic, too -- did you mean to keep the `&&` instead of changing it to `||`? maybe I misunderstood your slack message